crates/rpc/rpc-builder/src/lib.rs

```diff
@@ -1918,100 +1923,40 @@ impl Default for WsHttpServers {
     }
 }
 
-/// Http Servers Enum
+
 enum WsHttpServerKind {
-    /// Http server
-    Plain(Server<Identity, RpcServerMetrics>),
-    /// Http server with cors
-    WithCors(Server<Stack<CorsLayer, Identity>, RpcServerMetrics>),
-    /// Http server with auth
-    WithAuth(Server<Stack<AuthLayer<JwtAuthValidator>, Identity>, RpcServerMetrics>),
-    /// Http server with cors and auth
-    WithCorsAuth(
-        Server<Stack<AuthLayer<JwtAuthValidator>, Stack<CorsLayer, Identity>>, RpcServerMetrics>,
-    ),
+    Server(Server<tower::layer::util::Stack<tower::util::Either<reth_rpc::AuthLayer<reth_rpc::JwtAuthValidator>, tower::layer::util::Identity>, tower::layer::util::Stack<tower::util::Either<std::result::Result<tower_http::cors::CorsLayer, cors::CorsDomainError>, tower::layer::util::Identity>, tower::layer::util::Identity>>, metrics::RpcServerMetrics>),
```

Comment:
yeah this is a bit horrible, but better than before,

can convert this to a struct now